packages/@aws-cdk/toolkit/lib/toolkit.ts

```diff
@@ -462,9 +464,101 @@ export class Toolkit {
    * Continuously observe project files and deploy the selected stacks automatically when changes are detected.
    * Implies hotswap deployments.
    */
-  public async watch(_cx: ICloudAssemblySource, _options: WatchOptions): Promise<void> {
+  public async watch(cx: ICloudAssemblySource, options: WatchOptions): Promise<void> {
     const ioHost = withAction(this.ioHost, 'watch');
-    throw new Error('Not implemented yet');
+    const rootDir = options.rootDir ?? path.dirname(path.resolve(PROJECT_CONFIG));
```

Comment:
I don't think this line is doing what it claims to do `path.resolve` is just appending `cdk.json` to the CWD and then we take the dirname of that. Which is also just the CWD.